### PR TITLE
Refactor(eos_designs, eos_cli_config_gen): BGP VRF peer group options, global context ipv6 multicast and flowspec

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

This PR is refactoring peer group configuration under VRF mode in BGP. Since all the configurable options for peer groups under VRF submode get pushed out to the global mode, these options have been moved to the global BGP context where applicable.

New multicast ipv6 and flow-spec AFs have been implemented in the global context since there was an indirect way to configure them via the VRF submodes with the old (now deprecated) vrf.address_families data model.

Also featuring an update to router_bgp.address_family_ipv4.neighbor.<ip>.next_hop.address_family_ipv6_originate data model to add an option for enabling it without the "originate" keyword which is also supported in EOS.

## Component(s) name

`arista.avd.eos_cli_config_gen`
`arista.avd.eos_designs`

## Proposed changes

```yaml
router_bgp:
      ...
      address_family_ipv6_multicast:
        type: dict
        keys:
          bgp:
            type: dict
            keys:
              missing_policy:
                type: dict
                keys:
                  direction_in_action:
                    type: str
                    valid_values:
                    - deny
                    - deny-in-out
                    - permit
                  direction_out_action:
                    type: str
                    valid_values:
                    - deny
                    - deny-in-out
                    - permit
              additional_paths:
                type: dict
                keys:
                  receive:
                    type: bool
          neighbors:
            type: list
            primary_key: ip_address
            convert_types:
            - dict
            items:
              type: dict
              keys:
                ip_address:
                  type: str
                activate:
                  type: bool
                route_map_in:
                  type: str
                  description: Inbound route-map name
                route_map_out:
                  type: str
                  description: Outbound route-map name
          peer_groups:
            type: list
            primary_key: name
            convert_types:
            - dict
            items:
              type: dict
              keys:
                name:
                  type: str
                  description: Peer-group name
                activate:
                  type: bool
          networks:
            type: list
            primary_key: prefix
            convert_types:
            - dict
            items:
              type: dict
              keys:
                prefix:
                  type: str
                  description: IPv6 prefix "A:B:C:D:E:F:G:H/I"
                route_map:
                  type: str
      address_family_flow_spec_ipv4:
        type: dict
        keys:
          bgp:
            type: dict
            keys:
              missing_policy:
                type: dict
                keys:
                  direction_in_action:
                    type: str
                    valid_values:
                    - deny
                    - deny-in-out
                    - permit
                  direction_out_action:
                    type: str
                    valid_values:
                    - deny
                    - deny-in-out
                    - permit
          neighbors:
            type: list
            primary_key: ip_address
            convert_types:
            - dict
            items:
              type: dict
              keys:
                ip_address:
                  type: str
                activate:
                  type: bool
          peer_groups:
            type: list
            primary_key: name
            convert_types:
            - dict
            items:
              type: dict
              keys:
                name:
                  type: str
                  description: Peer-group name
                activate:
                  type: bool
      address_family_flow_spec_ipv6:
        type: dict
        keys:
          bgp:
            type: dict
            keys:
              missing_policy:
                type: dict
                keys:
                  direction_in_action:
                    type: str
                    valid_values:
                    - deny
                    - deny-in-out
                    - permit
                  direction_out_action:
                    type: str
                    valid_values:
                    - deny
                    - deny-in-out
                    - permit
          neighbors:
            type: list
            primary_key: ip_address
            convert_types:
            - dict
            items:
              type: dict
              keys:
                ip_address:
                  type: str
                activate:
                  type: bool
          peer_groups:
            type: list
            primary_key: name
            convert_types:
            - dict
            items:
              type: dict
              keys:
                name:
                  type: str
                  description: Peer-group name
                activate:
                  type: bool
```

## How to test

Expanded molecule to test the new knobs.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 1]
```diff
@@ -926,8 +926,13 @@ keys:
...
                   keys:
-                    address_family_ipv6_originate:
-                      type: bool
+                    address_family_ipv6:
+                      type: dict
+                      keys:
+                        enabled:
+                          type: bool
+                        originate:
+                          type: bool
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml:936]:
```suggestion
                          type: bool
                    address_family_ipv6_originate:
                      type: bool
                      deprecation:
                        warning: true
                        removed: true
                        new_key: address_family_ipv6
                        remove_in_version: 4.0.0
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 2]
```diff
@@ -926,8 +926,13 @@ keys:
                 next_hop:
                   type: dict
                   keys:
-                    address_family_ipv6_originate:
-                      type: bool
+                    address_family_ipv6:
+                      type: dict
+                      keys:
+                        enabled:
+                          type: bool
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml:933]:
```suggestion
                          type: bool
                          required: true
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml [hunk 3]
```diff
@@ -1597,31 +1764,6 @@ keys:
                             keys:
                               enabled:
                                 type: bool
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]:
```suggestion
                                type: bool
                                required: true
                peer_groups:
                  type: list
                  deprecation:
                    warning: true
                    removed: true
                    new_key: router_bgp.peer_groups
                    remove_in_version: 4.0.0
```

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Same for the other removed peer_groups

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Added some schema comments. Written in free-hand, so please verify that the models are correct. Please keep the `required: true` on the `enabled` key.
Please add release-notes and porting-guide details.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml (hunk 3 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml]: Have added back everywhere with deprecation warnings.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/router-bgp.yml [hunk 4]
```diff
@@ -38,6 +38,8 @@ router_bgp:
     peer_groups:
       EVPN-OVERLAY-PEERS:
         activate: true
+        next_hop:
+          address_family_ipv6_originate: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen_deprecated_vars/inventory/host_vars/host1/router-bgp.yml:43]:
```suggestion
          # Testing address_family_ipv6_originate which is deprecated in 4.0.0. To be removed in 5.0.0
          address_family_ipv6_originate: true
```

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**